[2023-04-23 08:20:00,587] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: multi_attack_dag.attack_ProjectedGradientDescent manual__2023-04-23T08:19:40.670843+00:00 [queued]>
[2023-04-23 08:20:00,602] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: multi_attack_dag.attack_ProjectedGradientDescent manual__2023-04-23T08:19:40.670843+00:00 [queued]>
[2023-04-23 08:20:00,603] {taskinstance.py:1356} INFO - 
--------------------------------------------------------------------------------
[2023-04-23 08:20:00,604] {taskinstance.py:1357} INFO - Starting attempt 1 of 2
[2023-04-23 08:20:00,605] {taskinstance.py:1358} INFO - 
--------------------------------------------------------------------------------
[2023-04-23 08:20:00,622] {taskinstance.py:1377} INFO - Executing <Task(PythonOperator): attack_ProjectedGradientDescent> on 2023-04-23 08:19:40.670843+00:00
[2023-04-23 08:20:00,631] {standard_task_runner.py:52} INFO - Started process 1566 to run task
[2023-04-23 08:20:00,635] {standard_task_runner.py:79} INFO - Running: ['***', 'tasks', 'run', 'multi_attack_dag', 'attack_ProjectedGradientDescent', 'manual__2023-04-23T08:19:40.670843+00:00', '--job-id', '5960', '--raw', '--subdir', 'DAGS_FOLDER/attack_dag.py', '--cfg-path', '/tmp/tmpq0idvpxj', '--error-file', '/tmp/tmp9jg64m3e']
[2023-04-23 08:20:00,636] {standard_task_runner.py:80} INFO - Job 5960: Subtask attack_ProjectedGradientDescent
[2023-04-23 08:20:00,695] {task_command.py:369} INFO - Running <TaskInstance: multi_attack_dag.attack_ProjectedGradientDescent manual__2023-04-23T08:19:40.670843+00:00 [running]> on host a3dd7d962104
[2023-04-23 08:20:00,781] {taskinstance.py:1571} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=multi_attack_dag
AIRFLOW_CTX_TASK_ID=attack_ProjectedGradientDescent
AIRFLOW_CTX_EXECUTION_DATE=2023-04-23T08:19:40.670843+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-04-23T08:19:40.670843+00:00
[2023-04-23 08:20:00,784] {transport.py:157} INFO - Attempting refresh to obtain initial access_token
[2023-04-23 08:20:00,788] {client.py:777} INFO - Refreshing access_token
[2023-04-23 08:20:01,857] {transport.py:157} INFO - Attempting refresh to obtain initial access_token
[2023-04-23 08:20:01,859] {client.py:777} INFO - Refreshing access_token
[2023-04-23 08:20:02,902] {taskinstance.py:1889} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 171, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 189, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/attack_dag.py", line 142, in attack_ProjectedGradientDescent
    model_acc,adversarial_examples = attack(ProjectedGradientDescent)
  File "/opt/airflow/dags/attack_dag.py", line 163, in attack
    estimator = get_estimator()
  File "/opt/airflow/dags/attack_dag.py", line 69, in get_estimator
    model = load_from_bucket("ML_model.pickle")
  File "/opt/airflow/dags/helpers.py", line 121, in load_from_bucket
    raise err
  File "/opt/airflow/dags/helpers.py", line 118, in load_from_bucket
    loaded_obj = pickle.load(f)
AttributeError: Can't get attribute 'NeuralNetworkClassificationModel' on <module '__main__' from '/home/***/.local/bin/***'>
[2023-04-23 08:20:02,911] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=multi_attack_dag, task_id=attack_ProjectedGradientDescent, execution_date=20230423T081940, start_date=20230423T082000, end_date=20230423T082002
[2023-04-23 08:20:02,922] {standard_task_runner.py:97} ERROR - Failed to execute job 5960 for task attack_ProjectedGradientDescent (Can't get attribute 'NeuralNetworkClassificationModel' on <module '__main__' from '/home/***/.local/bin/***'>; 1566)
[2023-04-23 08:20:02,941] {local_task_job.py:156} INFO - Task exited with return code 1
[2023-04-23 08:20:02,972] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
